src/types/time.rs

```diff
@@ -49,14 +121,15 @@ mod test {
 
         let mut ts_vec = vec![];
 
-        let make_datetime =
-            |secs, nanos| OffsetDateTime::from_unix_timestamp(secs) + Duration::from_nanos(nanos);
+        let make_datetime = |secs, millis| {
+            OffsetDateTime::from_unix_timestamp(secs) + Duration::from_millis(millis)
+        };
 
         ts_vec.push(make_datetime(10_000, 0)); //January 1, 1970 2:46:40 AM
-        ts_vec.push(make_datetime(10_000, 1000)); //January 1, 1970 2:46:40 AM (and one microsecond)
-        ts_vec.push(make_datetime(1_500_391_124, 1_000_000)); //July 18, 2017
```

Comment:
https://github.com/sqlite/sqlite/blob/master/src/date.c#L216
There seems to be no limit on the number of digits in subsecond.

Reply:
Thanks, don't know how I missed this. Seems like conversion FromSql and back ToSql could be lossy then?